sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateBoxDao.java

```diff
@@ -182,6 +184,61 @@ public BoxUse getUseById(long id) throws IOException {
     return results;
   }
 
+  @Override
+  public List<Box> getByPartialSearch(String search) {
+    if (search == null) {
+      throw new NullPointerException("No search String provided");
+    }
+    Criteria criteria = currentSession().createCriteria(BoxImpl.class);
+    criteria.add(Restrictions.or(
+        Restrictions.like("identificationBarcode", search, MatchMode.ANYWHERE),
+        Restrictions.like("name", search, MatchMode.ANYWHERE),
+        Restrictions.like(FIELD_ALIAS, search, MatchMode.ANYWHERE)));
+    @SuppressWarnings("unchecked")
+    List<Box> results = criteria.list();
+    results.sort(new Comparator<Box>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)